Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -345,6 +354,20 @@ private void OnValidateCommand(ValidateCommandEvent evt)
             }
         }
 
+        private void OnPointerDown(PointerDownEvent evt)
+        {
+            // Allow right clicks to select an item before we bring up the matching context menu.
+            if (evt.button == (int)MouseButton.RightMouse && evt.clickCount == 1)
+            {
+                // Look upwards to the immediate child of the scroll view, so we know what Index to use
+                var element = evt.target as VisualElement;
+                while (element.name != "unity-tree-view__item")
+                    element = element.parent;
```

Comment:
how about use panel.Pick(evt.mousePosition)? that could save a while loop here

Reply:
I tried that - unfortunately it just returns the same element as evt.target 😞
I think the issue is that in the scrollview, the hierarchy _should_ have everything descending from InputActionsTreeViewItem, which would mean I could get ancestor of type...

Reply:
right, that would be a better option. If this is not working we should at least introduce a safety break in this loop, if no element is found. Something like element.parent !=null

Reply:
I _think_ in theory that shouldn't be possible, but sensible idea - added in b9f698fd36f6995569cb232ba3e6d26e0d1820bf.